src/deprecated-rules.spec.ts

```diff
@@ -0,0 +1,107 @@
+// deprecated-rules.spec.ts
+
+import { strict as assert } from 'node:assert';
+import { describe, it } from 'node:test';
+
+// eslint-disable-next-line sonarjs/deprecation
+import { builtinRules } from 'eslint/use-at-your-own-risk';
+import { rules as sonarjsRules } from 'eslint-plugin-sonarjs';
```

Comment:
obviously I don't know the details, but it seems weird we would need to import or have knowledge of specific rule sets defined in `index.mjs`.  Can we do this based purely on `index.mjs`, and if we need extra stuff we dynamically import based on that data?  Or is it simply not possible.

Reply:
i remember that we discussed an idea to create a instance of Eslint and explore what rules are effective after they are calculated. i tried that and found sth interesting:
```
    const linter = new ESLint({
      baseConfig: config,
    });

    const results = await linter.lintFiles('./src/index.ts');
    console.log('deprecated rules', results[0]?.usedDeprecatedRules);
```
it turns out we don't have to target/parse specific rule sets since the Eslint can tell us which deprecated rules are enabled. this should make it much simpler and safer. all we need is  to check which ones are specifically enabled in our config.